// @clerk/clerk-sdk-node/dist/cjs/instance.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var instance_exports = {};
__export(instance_exports, {
  AllowlistIdentifier: () => import_backend.AllowlistIdentifier,
  Client: () => import_backend.Client,
  Email: () => import_backend.Email,
  EmailAddress: () => import_backend.EmailAddress,
  ExternalAccount: () => import_backend.ExternalAccount,
  IdentificationLink: () => import_backend.IdentificationLink,
  Invitation: () => import_backend.Invitation,
  Organization: () => import_backend.Organization,
  OrganizationInvitation: () => import_backend.OrganizationInvitation,
  OrganizationMembership: () => import_backend.OrganizationMembership,
  OrganizationMembershipPublicUserData: () => import_backend.OrganizationMembershipPublicUserData,
  PhoneNumber: () => import_backend.PhoneNumber,
  SMSMessage: () => import_backend.SMSMessage,
  Session: () => import_backend.Session,
  User: () => import_backend.User,
  Verification: () => import_backend.Verification,
  default: () => instance_default
});
module.exports = __toCommonJS(instance_exports);
var import_clerkClient = require("./clerkClient");
var import_backend = require("@clerk/backend");
var instance_default = import_clerkClient.Clerk;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AllowlistIdentifier,
  Client,
  Email,
  EmailAddress,
  ExternalAccount,
  IdentificationLink,
  Invitation,
  Organization,
  OrganizationInvitation,
  OrganizationMembership,
  OrganizationMembershipPublicUserData,
  PhoneNumber,
  SMSMessage,
  Session,
  User,
  Verification
});
